File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.24-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function pattern_24() {
    let row = 10;
    let column = 10;

    for (let y = 1; y < row + 1; y++) {
        for (let x = 1; x <= column; x++) {
            if (x <= column / 2) {
                if (y <= row / 2) {
                    if (x <= row / 2 - y + 1) {
                        log(x);
                    } else {
                        log(' ');
                    }
                } else if (y >= row / 2) {
                    if (x <= y - row / 2) {
                        log(x);
                    } else {
                        log(' ');
                    }
                }
            } else if (x > column / 2) {
                let key = column - x + 1;
                if (y <= row / 2) {
                    if (key <= row / 2 - y + 1) {
                        log(key);
                    } else {
                        log(' ');
                    }
                } else if (y >= row / 2) {
                    if (key <= y - row / 2) {
                        log(key);
                    } else {
                        log(' ');
                    }
                }
            }
        }

        console.log(' ', y);
    }

    console.log();
    return 0;
}

pattern_24();

File: python/0.0-Basics/0.3-build-up-logical-thinking/0.3.24-pattern.py
from utils.clear import clear

def pattern_24():
    row = 10
    column = 10

    for i in range(row , -1, -1):
        for j in range(0, column + 1):
            if j <= column / 2:
                if i >= row / 2:
                    key = round(i - row / 2)
                    if j <= key:
                        print(j, end="")
                    elif j > key:
                        print(" ", end="")
                elif i < row / 2:
                    key = round( row / 2  - i)
                    if j <= key:
                        print(j, end="")
                    elif j > key:
                        print(" ", end="")

            elif j > column / 2:
                newJ = column - j

                if i >= row / 2 :
                    newI = round(i - row / 2)
                    if newJ <= newI :
                        print(newJ, end="")
                    elif newJ > newI:
                        print(" ", end="")
                elif i < row / 2:
                    newI = round( row / 2  - i)
                    if newJ <= newI:
                        print(newJ, end="")
                    elif newJ > newI:
                        print(" ", end="")


        print()


clear()
pattern_24()
